### ISSUE TITLE
Unable to view dhat-heap.json - "data file is missing a field: mi"

### ISSUE THREAD

**Issue by gregtatum** [nnethercote/dhat-rs#10]
Steps to reproduce:

* Add `dhat = "0.2.1"` to a project
* Create and run a simple `cargo run --example`.

```
➤ cargo run --example memory
    Finished dev [unoptimized + debuginfo] target(s) in 0.09s
     Running `/Users/greg/dev/icu4x/target/debug/examples/memory`
dhat: Total:     20,122 bytes in 129 blocks
dhat: At t-gmax: 9,328 bytes in 90 blocks
dhat: At t-end:  0 bytes in 0 blocks
dhat: The data in dhat-heap.json is viewable with dhat/dh_view.html
```

* I'm not sure at this point what dhat/dh_view.html is referring to.
* It's not located in the current directory.
* Download https://www.valgrind.org/downloads/current.html 
* Open file:///Users/greg/Downloads/valgrind-3.16.1/dhat/dh_view.html
* Click load button, and load `dhat-heap.json`

```
Uncaught Error: data file is missing a field: mi
    checkFields http://localhost:8000/dh_view.js:472
    buildTree http://localhost:8000/dh_view.js:509
    onload http://localhost:8000/dh_view.js:1351
    tryFunc http://localhost:8000/dh_view.js:265
    onload http://localhost:8000/dh_view.js:1341
    loadFile http://localhost:8000/dh_view.js:1340
```

* Also try on `git clone git://sourceware.org/git/valgrind.git`
* The error still reproduces.

**Comment by gregtatum** [comment]
Here is the JSON from my run:

[dhat-heap.json.zip](https://github.com/nnethercote/dhat-rs/files/5732140/dhat-heap.json.zip)

**Comment by gregtatum** [comment]
Ah, so it turns out I had a copy of 3.16. I followed the documentation more closely, and figured out the mismatch there that 3.17 is required. I think I read the docs out of order. I'll leave this open for now, as it my be worth clarifying this somehow either in the docs, or in the printed text.

I also filed #11 , which would help with this.

Additionally, the current release is 3.16.1, which is not valid for this tool.
